compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -8363,7 +8366,8 @@ private STNode parseTypeTestExpression(STNode lhsExpr, boolean isInConditionalEx
      */
     private STNode parseIsKeyword() {
         STToken token = peek();
-        if (token.kind == SyntaxKind.IS_KEYWORD) {
+        if (token.kind == SyntaxKind.IS_KEYWORD ||
+                token.kind == SyntaxKind.NOT_IS_KEYWORD) {
```

Comment:
In order for the error handler logic to be aligned with this, we need to add an entry for `NOT_IS_KEYWORD` there.

Currently, as per the below logic `hasMatch` will only be true for `IS_KEYWORD` kind for `ParserRuleContext.IS_KEYWORD`.
https://github.com/ballerina-platform/ballerina-lang/blob/c2fde7f1422a22d547041be4eed1a8a9bcd6aa29/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java#L1171-L1172

We need to add an entry similar to this.
https://github.com/ballerina-platform/ballerina-lang/blob/c2fde7f1422a22d547041be4eed1a8a9bcd6aa29/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java#L1112-L1115

Reply:
May be change the method name to `parseIsOrNotIsKeyword` and update the doc comment accordingly.
Update the doc comment above `parseTypeTestExpression` as well for the new rule.

Reply:
Test case:
```
if a ! !is int {
        
}
```
should be recovered as,
```
if a  INVALID[!] !is int {
        
}
```